### ISSUE TITLE
Add/update documentation

### ISSUE THREAD

**Issue by ras44** [uber/causalml#677]
During Friday's office hours we discussed adding an introduction for new users about meta-learners, propensity modeling, etc.  This issue was created to track and collaborate on those updates.

@jeongyoonlee, a couple questions:
- just to confirm, should we focus on edits to the content in docs/*.rst?
- any specific areas that you feel are in need of attention?

Thanks!

**Comment by ras44** [comment]
Some notes on areas that could possibly use attention:

- About CausalML (https://causalml.readthedocs.io/en/latest/about.html)
  - Is it `Causal ML` or `CausalML`?
  - use `W` for treatment variable for notation consistency
  - Add material on Neural network based algorithms
    - CEVAE
    - dragonnet

- Methodology
  - Perhaps add a section on propensity score, since it is referenced frequently?
  - S-Learner
    - Stage 2
      - Here we fit the model with the entire dataset, including the treatment flag as a covariate
      - Possibly modify: 'Including the *treatment* propensity score...`
      - Perhaps "When the control and treatment groups have very different outcomes with respect to the covariates..."
  - T-Learner
    - In other words: we fit two models, each using the data from one of the two possible outcomes
    - define response under control, response under treatment \mu_0 and \mu_1
  - X-Learner
    - perhaps a high level intro mentioning the fact that X-Learners generate two estimates for CATE and then take a weighted average
  - R-Learner
    - add detail from Nie and Wager's paper

... will continue to add

**Comment by jeongyoonlee** [comment]
Thanks for taking the lead on this, @ras44. 

To answer some of the questions:
- `docs/*.rst` will be our main documentation published to [readthedocs](https://causalml.readthedocs.io/), while `README.md` will host a subset of instructions for installation and quick start examples.
- Let's use `CausalML` to refer to the package.
- I agree that we should use consistent variable naming conventions. e.g., `W` for treatment assignment

I will add more later.

**Comment by rolandrmgservices** [comment]
https://econml.azurewebsites.net/spec/flowchart.html

https://docs.readthedocs.io/en/stable/guides/jupyter.html

**Comment by paullo0106** [comment]
@jeongyoonlee reference to the User guide Flowchart we created in KDD 2021 tutorial, inspired by [EconML's flowchart](https://econml.azurewebsites.net/spec/flowchart.html)

![external  KDD 2021 - Introduction to CausalML (1)](https://github.com/uber/causalml/assets/4966393/2f017b54-33ca-48fb-a078-d0282274b4b9)
![external  KDD 2021 - Introduction to CausalML](https://github.com/uber/causalml/assets/4966393/0f2ca6da-ef54-4526-8edf-eef30a33fa9f)


* Source: `KDD 2021 KDD 2021 - Introduction to CausalML` - page 41 and 42 in [this slide](https://docs.google.com/presentation/d/1D-cqqwKyWseVoNQEH-d0TS9wqK55l3bSn3PaLFxNEWI/edit?usp=sharing), the slide [doesn't look public anymore](https://causal-machine-learning.github.io/kdd2021-tutorial/) maybe due to updates in Uber org's gsuite permission

**Comment by ras44** [comment]
hi @paullo0106 would it be possible to share a higher-res/sharper version of that second screenshot?  I was hoping to add it to the documentation 🙏

**Comment by ImanEmtiazi728** [comment]
I have a dataset containing two variables. I want to estimate the average treatment effect (CATE) using cross-validation between them, based on different learners, R, X, T, and S, to compare the outcomes. in my dataset, X and y are non-binary variables. I'm applying the https://github.com/uber/causalml/blob/master/docs/examples/meta_learners_with_synthetic_data.ipynb on my dataset. 
I attached the dataset to this comment. I want to define 'SystemLoadEP2' in my dataset as X and SMPEP2 as y. 
[Prices.csv](https://github.com/uber/causalml/files/13758757/Prices.csv)
Would you please help with it?

**Comment by ImanEmtiazi728** [comment]
I have a dataset containing two variables. To compare the outcomes, I want to estimate ATE and CATE using cross-validation between them, based on different learners, R, X, T, and S. in my dataset, X and y are non-binary variables. I'm applying the https://github.com/uber/causalml/blob/master/docs/examples/meta_learners_with_synthetic_data.ipynb on my dataset.
I attached the dataset to this comment. I want to define 'SystemLoadEP2' in my dataset as X and SMPEP2 as y.
Would you please help with it?
[Prices.csv](https://github.com/uber/causalml/files/13758764/Prices.csv)
